### ISSUE TITLE
BP user check can generate PHP errors

### ISSUE THREAD

**Issue by boonebgorges** [johnbillion/user-switching#23]
The BuddyPress current user checks at https://github.com/johnbillion/user-switching/blob/3b9ec9cc25b6bad039878eeaddb9a5d7e3065480/user-switching.php#L617 can cause PHP notices when not looking at a single user profile *or* a member directory. I think what's needed is something like this:

```php
$user_id = null;
if ( bp_is_user() ) {
    $user_id = bp_displayed_user_id();
} elseif ( bp_is_members_directory() ) {
    $user_id = bp_get_member_user_id();
}

if ( ! $user_id ) {
   return;
}

$user = get_userdata( $user_id );

// ...

```

**Comment by johnbillion** [comment]
Cheers. This is code that pre-dates those functions (it's existed since something like BuddyPress 1.3). I'm pleased that it mostly still works, but I'll update it to use the template functions instead.

**Comment by johnbillion** [comment]
Oh that's right, there wasn't a 1.3. Haha. But yeah, I'll fix this.

**Comment by johnbillion** [comment]
Fixed in 1.3.0.
